### PR TITLE
Add Go SDK and toolchain for Terratest

### PR DESCRIPTION
**Pull request by foreverXZC**

### CODE REVIEW

File: tool/env_setup.sh [hunk 1]
```diff
@@ -67,18 +69,25 @@ curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform
...
 gem update --system && \
-  gem install bundler --no-document --version=${GEM_BUNDLER_VERSION} && \
-  gem install colorize --no-document --version=${GEM_COLORIZE} && \
-  gem install kitchen-terraform --no-document --version=${GEM_KITCHEN_TERRAFORM_VERSION} && \
-  gem install test-kitchen --no-document --version=${GEM_TEST_KITCHEN_VERSION} && \
-  gem install rake --no-document --version=${GEM_RAKE_VERSION} && \
-  gem install rspec --no-document --version=${GEM_RSPEC_VERSION}
-DOUBLE DEALS
-PROFILE
+  gem install bundler --no-document --version=${GEM_BUNDLER_VERSION} --user-install && \
```
Comment (metacpp) [inline tool/env_setup.sh:85]: why did you add --user-install?

File: tool/env_setup.sh [hunk 2]
```diff
@@ -67,18 +69,25 @@ curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform
...
+  gem install colorize --no-document --version=${GEM_COLORIZE} --user-install && \
+  gem install kitchen-terraform --no-document --version=${GEM_KITCHEN_TERRAFORM_VERSION} --user-install && \
```
Comment (metacpp) [inline tool/env_setup.sh]: I don't think we still need kitchen-terraform, test-kitchen. For rspec, maybe we also can remove it.

File: tool/env_setup.sh [hunk 3]
```diff
@@ -42,7 +44,7 @@ else
             libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 \
             libgdbm-dev unzip
     git clone https://github.com/rbenv/rbenv.git ~/.rbenv
-    export PATH="$HOME/.rbenv/bin:$PATH"
+    export PATH="$HOME/.rbenv/bin:$HOME/.rbenv/versions/${RUBY_VERSION}/bin:$PATH"
```
Comment (metacpp) [inline tool/env_setup.sh]: Why did you add "HOME/.rbenv/versions/${RUBY_VERSION}/bin"? If $RBENV_VERSION is set, rbenv will automatically select that version of ruby.

File: tool/env_setup.sh [hunk 4]
```diff
@@ -67,18 +69,25 @@ curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform
...
   shasum -a 256 -c terraform_${TERRAFORM_VERSION}_SHA256SUMS 2>&1 | grep "${TERRAFORM_VERSION}_${TERRAFORM_OS_ARCH}.zip:\sOK" && \
   unzip -o terraform_${TERRAFORM_VERSION}_${TERRAFORM_OS_ARCH}.zip -d /usr/local/bin
-# Cleanup downloaded files
+# Cleanup downloaded files.
 rm terraform_${TERRAFORM_VERSION}_${TERRAFORM_OS_ARCH}.zip
 rm terraform_${TERRAFORM_VERSION}_SHA256SUMS
 rm terraform_${TERRAFORM_VERSION}_SHA256SUMS.sig
 
+# Install Go SDK.
+curl -Os https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-amd64.tar.gz >/dev/null 2>&1 && \
```
Comment (metacpp) [inline tool/env_setup.sh]: Please refer to the install script of terraform(line#4). This script should support both linux and darwin.

File: tool/env_setup.sh [hunk 5]
```diff
@@ -67,18 +69,25 @@ curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform
...
+  gem install bundler --no-document --version=${GEM_BUNDLER_VERSION} --user-install && \
+  gem install colorize --no-document --version=${GEM_COLORIZE} --user-install && \
+  gem install kitchen-terraform --no-document --version=${GEM_KITCHEN_TERRAFORM_VERSION} --user-install && \
+  gem install test-kitchen --no-document --version=${GEM_TEST_KITCHEN_VERSION} --user-install && \
+  gem install rake --no-document --version=${GEM_RAKE_VERSION} --user-install && \
+  gem install rspec --no-document --version=${GEM_RSPEC_VERSION} --user-install
+
+# install go dep
+export GOPATH="$HOME/go"
+export PATH="$HOME/.rbenv/bin:$HOME/.rbenv/versions/${RUBY_VERSION}/bin:/usr/local/go/bin:$GOPATH/bin:$PATH"
```
Comment (metacpp) [inline tool/env_setup.sh]: Why did you append ruby path again?

File: tool/env_setup.sh [hunk 6]
```diff
@@ -67,18 +69,25 @@ curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform
...
-DOUBLE DEALS
-PROFILE
+  gem install bundler --no-document --version=${GEM_BUNDLER_VERSION} --user-install && \
+  gem install colorize --no-document --version=${GEM_COLORIZE} --user-install && \
+  gem install kitchen-terraform --no-document --version=${GEM_KITCHEN_TERRAFORM_VERSION} --user-install && \
+  gem install test-kitchen --no-document --version=${GEM_TEST_KITCHEN_VERSION} --user-install && \
+  gem install rake --no-document --version=${GEM_RAKE_VERSION} --user-install && \
+  gem install rspec --no-document --version=${GEM_RSPEC_VERSION} --user-install
+
+# install go dep
```
Comment (metacpp) [inline tool/env_setup.sh]: Wrong comment, below 2 lines of code is used to refresh the env for Go, only last LOC is for Go dep.

**Review by metacpp — CHANGES_REQUESTED** [review]

File: tool/env_setup.sh (hunk 1 above)
Comment (foreverXZC) [reply tool/env_setup.sh:85]: Because I was not permitted to install these packages unless I add --user-install.

File: tool/env_setup.sh (hunk 2 above)
Comment (foreverXZC) [reply tool/env_setup.sh]: I will remove kitchen, but I think rspec should be kept because we are still using task.

File: tool/env_setup.sh (hunk 3 above)
Comment (foreverXZC) [reply tool/env_setup.sh]: I remember that if I remove this part, ruby will not be found. Let me try again.

File: tool/env_setup.sh (hunk 4 above)
Comment (foreverXZC) [reply tool/env_setup.sh]: Sounds good.

File: tool/env_setup.sh (hunk 5 above)
Comment (foreverXZC) [reply tool/env_setup.sh]: Will remove ruby.

File: tool/env_setup.sh (hunk 6 above)
Comment (foreverXZC) [reply tool/env_setup.sh]: Will change comment.

**Review by metacpp — APPROVED** [review]
LGTM.
